service/history/api/startworkflow/api.go

```diff
@@ -348,6 +347,11 @@ func (s *Starter) createAsCurrent(
 	creationParams *creationParams,
 	currentWorkflowConditionFailed *persistence.CurrentWorkflowConditionFailedError,
 ) error {
+	// TODO(stephanos): remove this hack
+	// This is here for Update-with-Start to reapply the Update after it was aborted previously.
+	if _, err := s.createOrUpdateLeaseFn(creationParams.workflowLease, s.shardContext, nil); err != nil {
+		return err
+	}
```

Comment:
Note that his will now call NewWorkflowLeaseAndContext twice for a regular workflow start. But that is fine, as I added a special case to make that a noop there.

Reply:
do you mean `creationParams.workflowLease` is always nil?